internal/rundata/rundata.go

```diff
@@ -125,6 +138,21 @@ func Properties(ctx context.Context, resv *binding.Reservation) map[string]strin
 	return m
 }
 
+// readMetadataProto reads and unmarshals the metadata proto file.
+// If a metadata proto does not exist, returns a nil metadata and nil error.
+func readMetadataProto() (*mpb.Metadata, error) {
+	const metadataFilename = "metadata.textproto"
+	if _, err := os.Stat(metadataFilename); err != nil {
```

Comment:
While assuming that metadata proto exists at  ./metadata.textproto may work for third_party_go_test, this won't work for open source developers because they typically run a test at repo root like this:

```sh
go test ./feature/subfeature/otg_tests/foo_test
```

The test might also pick up the wrong metadata file if it happens to run from the wrong directory.

The primary mechanism should be use the source file path to find the metadata. We can factor out the sourceFullPath from these lines:

https://github.com/openconfig/featureprofiles/blob/bd74c0dc59bb5866677084f3298d20191ee0946c/internal/rundata/local.go#L154-L168

A secondary mechanism for Bazel could be to use os.Args[0] to find the test executable path, and replace the executable's basename with metadata.textproto. This should work for both go_test and third_party_go_test.

Reply:
Actually, it will work if someone runs `go test ./feature/subfeature/otg_tests/foo_test`. Go test always uses the test binary (the test package path) as the current working directory.